tests/pytests/unit/utils/parsers/test_log_parsers.py

```diff
@@ -0,0 +1,775 @@
+):  # pylint: disable=invalid-name
+    """
+    Tests that both console log level and log file level setting are working together
+    """
+    log_level = "critical"
+    log_level_logfile = "debug"
+
+    args = ["--log-file-level", log_level_logfile] + args
+
+    opts = testing_config.copy()
+    opts.update({loglevel_config_setting_name: log_level})
+
+    parser = parser()
+    with patch(config_func, MagicMock(return_value=opts)):
+        parser.parse_args(args)
+
+    log_level_logfile_option = getattr(
+        parser.options, logfile_loglevel_config_setting_name
+    )
+
+    # Check console logger
+    assert log_impl.log_level_console == log_level
+    # Check extended logger
+    assert log_impl.config[loglevel_config_setting_name] == log_level
+    assert log_impl.config[logfile_loglevel_config_setting_name] == log_level_logfile
+    # Check temp logger
+    assert log_impl.temp_log_level == "error"
+    # Check log file level setting
+    assert log_level_logfile_option == log_level_logfile
+    # Check log file logger
+    assert log_impl.log_level_logfile == log_level_logfile
+
+
+def test_log_created(
+    testing_config, args, parser, config_func, logfile_config_setting_name, log_file
+):
+    """
+    Tests that log file is created
+    """
+    opts = testing_config.copy()
+    opts.update({"log_file": str(log_file)})
+    log_file_name = str(log_file)
+    if log_file_name.split(os.sep)[-1] != "log_file":
+        opts.update({log_file_name: str(log_file)})
+
+    parser = parser()
+    with patch(config_func, MagicMock(return_value=opts)):
+        parser.parse_args(args)
+
+    assert os.path.exists(str(log_file_name))
+
+
+def test_callbacks_uniqueness(parser):
+    """
+    Test that the callbacks are only added once, no matter
+    how many instances of the parser we create
+    """
+    mixin_container_names = (
+        "_mixin_setup_funcs",
+        "_mixin_process_funcs",
+        "_mixin_after_parsed_funcs",
+        "_mixin_before_exit_funcs",
+    )
+    _parser = parser()
+    nums_1 = {}
+    for cb_container in mixin_container_names:
+        obj = getattr(_parser, cb_container)
+        nums_1[cb_container] = len(obj)
+
+    # The next time we instantiate the parser, the counts should be equal
+    _parser = parser()
+    nums_2 = {}
+    for cb_container in mixin_container_names:
+        obj = getattr(_parser, cb_container)
+        nums_2[cb_container] = len(obj)
+    assert nums_1 == nums_2
+
+
+def test_verify_log_warning_logged(args, config_func, testing_config, parser):
+    args = ["--log-level", "debug"] + args
+    with TstSuiteLoggingHandler(level=logging.DEBUG) as handler:
+        parser = parser()
+        with patch(config_func, MagicMock(return_value=testing_config)):
+            parser.parse_args(args)
+        assert (
+            "WARNING:Insecure logging configuration detected! Sensitive data may be logged."
+            in handler.messages
+        )
```

Comment:
```suggestion
def test_verify_log_warning_logged(args, config_func, testing_config, parser):
    args = ["--log-level", "debug"] + args
    with TstSuiteLoggingHandler(level=logging.DEBUG) as handler:
        parser = parser()
        with patch(config_func, MagicMock(return_value=testing_config)):
            parser.parse_args(args)
        assert (
            "WARNING:Insecure logging configuration detected! Sensitive data may be logged."
            in handler.messages
        )
```
Can we achieve the same with `caplog` and delete `TstSuiteLoggingHandler` if it's not getting used anywhere else?